packages/data/src/onboarding/resolvers.js

```diff
@@ -60,14 +68,77 @@ export function* getTasksStatus() {
 	}
 }
 
+function getQuery() {
+	const searchString = window.location && window.location.search;
+	if ( ! searchString ) {
+		return {};
+	}
+
+	const search = searchString.substring( 1 );
+	return parse( search );
+}
+
+/**
+ * This function will be depreciated in favour of registering tasks on the back-end.
+ *
+ * @param {Array} taskLists array of task lists.
+ */
+function* mergeWithFilteredTasks( taskLists ) {
+	const filteredTasks = applyFilters(
+		'woocommerce_admin_onboarding_task_list',
+		[],
+		getQuery()
+	);
+	if ( filteredTasks && filteredTasks.length > 0 ) {
+		deprecated( 'woocommerce_admin_onboarding_task_list', {
+			version: '2.10.0',
+			alternative: 'TaskLists::add_task()',
+			plugin: '@woocommerce/data',
+		} );
+		// Show depreciation notice;
+		const dismissedTasks = yield resolveSelect(
+			OPTIONS_STORE_NAME,
+			'getOption',
+			'woocommerce_task_list_dismissed_tasks'
+		) || [];
+		const snoozedTasks = yield resolveSelect(
+			OPTIONS_STORE_NAME,
+			'getOption',
+			'woocommerce_task_list_remind_me_later_tasks'
+		) || {};
+
+		// Format the old task items with the new format.
+		for ( const task of filteredTasks ) {
+			task.level = task.level ? parseInt( task.level, 10 ) : 3;
```

Comment:
Where is `level` used?  I'm not sure if we've used it in the new tasks- should this be added elsewhere?

Reply:
`level` is only supported in extended task list items, level 1 shows a red bar on the left and level 2 a yellow bar. They should also be sorted by level, level 1 being at the top.
See the screenshots here in wcpay for an example (https://github.com/Automattic/woocommerce-payments/pull/2922)